packages/core/src/application.ts

```diff
@@ -133,6 +139,9 @@ export class Application extends Context {
   private _setupOperation(verb: string, path: string, spec: OperationObject) {
     const handler = spec['x-operation'];
     if (typeof handler === 'function') {
+      spec = Object.assign({}, spec);
```

Comment:
Does this need to be a deep copy? https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Deep_Clone